### PR TITLE
Drop Cookie header and compress token payloads

### LINKED ISSUE #11301: [BUG] Bokeh websocket connection creates large Sec-WebSocket-Protocol header
Bokeh 2021.05

Currently the Cookie header is being included in the header, as well as an extra 'cookies' section which is a copy.

All cookies are whitelisted by default, as are all headers. 

When people have a bunch of cookies this can create a header so large common webservers reject it, in our application's case the python `websockets` library.

The data would also probably compress ok, but if the cookies doubling bug is fixed and it's changed to exclude cookies/headers by default instead of include this problem should be lessened dramatically.

### PR DESCRIPTION
**Pull request by philippjfr**
Implements the suggestions in #11301:

1. Drop the 'Cookie' header and restore it from the explicit Cookie payload.
2. Enable zlib compression for token payloads

The extra payloads are compressed separately from the main token content since the frontend has to access part of the token to get access to the session id. In the particular very bare-bones case I tested the compression only reduced the payload by about 10%, but even that is worth it.

- [x] issues: fixes #11301
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
cc @shughes-uk

**Comment by shughes-uk** [comment]
We see users with a ton of third party cookies who are pushing the 4096 limit just on their cookie header, so absolutely worth the compression

File: bokeh/util/token.py [hunk 1]
```diff
@@ -147,6 +150,9 @@ def get_token_payload(token: str) -> Any:
         dict
     """
     decoded = json.loads(_base64_decode(token.split('.')[0]))
+    if '__zlib__' in decoded:
+        decompressed = zlib.decompress(_base64_decode(decoded['__zlib__']))
+        decoded.update(json.loads(decompressed))
```
Comment (bryevdv) [inline bokeh/util/token.py:155]: shouldn't we `del decoded['__zlib__']` here for good measure? We might also want to document that `__zlib__` is a forbidden key (and also check explicitly for it above during encoding and raise a meaningful error)

Comment (philippjfr) [reply bokeh/util/token.py:155]: Yep, that all makes sense. Will try to finish this off this weekend.

**Comment by bryevdv** [comment]
>  removed this from the 2.4 milestone 9 days ago

@mattpap this is causing problems for one of out largest and most visible users right now. Unless you are proposing to fix this even sooner in a 2.3.x release then the 2.4 milestone must be restored.

**Comment by mattpap** [comment]
I removed the milestone, because it's already set on the issue.

**Comment by bryevdv** [comment]
> I removed the milestone, because it's already set on the issue.

We had a discussion where it was decided to make an explicit change from past practice, and always set milestones on PRs and we have been doing this for several months now. Are you proposing to change (back)? If so, please open a discussion.

**Comment by philippjfr** [comment]
Probably not before Friday.

**Comment by bryevdv** [comment]
@philippjfr @mattpap please have a quick look. I made a few additional small changes, including simplifying `_base64_decode` to always return only bytes (the "encoding" param was only used in tests so I made a helper function in the test with a different name for better code grepping)

**Comment by bryevdv** [comment]
Just noting, I'll plan to merge this tomorrow am PST unless more time is requested.

**Comment by mattpap** [comment]
Looks good to me (as far I can review this).

**Pull request merged**